### ISSUE TITLE
Parameters at limit not recognized

### ISSUE THREAD

**Issue by alexander-held** [scikit-hep/iminuit#483]
Hi, #469 introduced a summary table field that warns about parameters fitted at their limits. The following example does not produce a warning:

```python
import iminuit

def f(x):
    return (x - 1.2) ** 2

m = iminuit.Minuit(f, limit_x=(-1, 1), errordef=1, print_level=1)
m.migrad()
print(m.values)
print(f"at limit: {m.fmin.has_parameters_at_limit}")
```

with `iminuit` 1.5.0:

```
┌──────────────────────────────────┬──────────────────────────────────────┐
│ FCN = 0.04001                    │        Ncalls = 21 (21 total)        │
│ EDM = 1.14e-05 (Goal: 0.0002)    │               up = 1.0               │
├───────────────┬──────────────────┼──────────────────────────────────────┤
│ Valid Minimum │ Valid Parameters │        No Parameters at limit        │
├───────────────┴──────────────────┼──────────────────────────────────────┤
│          Below EDM goal          │           Below call limit           │
├───────────────┬──────────────────┼───────────┬─────────────┬────────────┤
│   Hesse ok    │  Has Covariance  │ Accurate  │  Pos. def.  │ Not forced │
└───────────────┴──────────────────┴───────────┴─────────────┴────────────┘
<ValueView of Minuit at 7f9a054bc740>
  x: 0.9999714990887609
at limit: False
```

The best-fit result of `x` should be 1.2, as confirmed by changing the limit to `(-1, 2)`.

This does not seem intended, unless I misunderstand the purpose of `has_parameters_at_limit`.

**Comment by HDembinski** [comment]
Thank you for provide this minimal example. You got the intend correctly, however, it is not clear how to define the threshold for "being at the limit". The current algorithm looks for deviation less < 1 sigma. Apparently this fails here because the uncertainty is too small. In the examples I tried, it seemed to work. Apparently I have to investigate more cases to come up with a better algorithm for detecting this.

**Comment by HDembinski** [comment]
On second glance, this looks like a bug. I am investigating.

**Comment by HDembinski** [comment]
fixed in #484
